### ISSUE TITLE
Rule proposal: arrow-spacing

### ISSUE THREAD

**Issue by inikulin** [eslint/eslint#2891]
Rule will enforce spacing around arrow in arrow functions. The rule is enabled by default.

**Options:**
The rule takes one option, an object, which has two keys before and after having boolean values true or false. If before is true, space is enforced before arrows and if it's false, space is disallowed before arrows. If after is true, space is enforced after arrows and if it's false, space is disallowed after arrows. The default is `{"before": true, "after": true}`.

**Example**:
  
```js
{ "before": true, "after": true }
```

The following patterns are considered warnings for this settings:
```js
(x) =>x * x
(x)=> x * x 
(x)=>x * x
```

The following patterns are not warnings for this settings:
```js
(x) => x * x
```

I can work on this rule if it will be accepted.

**Comment by ilyavolodin** [comment]
Duplicate of #2628 Closing this one as #2628 is more complete.
